src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -961,25 +961,79 @@ def _match_host_names_from_cert(hostnames_from_cert, hostnames_in_webapp):
     return matched
 
 
-def create_function(resource_group_name, new_app_name, storage_account, plan=None):
+def create_function(resource_group_name, name, storage_account, plan=None,
+                    consumption_plan_location=None):
+
+    if plan is not None and consumption_plan_location is not None:
+        raise CLIError("Use Plan or Consumption Plan Location")
```

Comment:
To clarify, validator gets wired up by binding it to any involved argument, and inside you can manipulate other arguments exposed from the `namespace` instance. 
Since your validation logic only applies on `create`, imo, it is fine not going through a validator, as long as you emit out the clear error Travis suggested